### ISSUE TITLE
Should cuDNN really be a requirement to import Lasagne?

### ISSUE THREAD

**Issue by JackKelly** [benanne/Lasagne#126]
In the latest version of Lasagne, I cannot `import lasagne` because I don't have cuDNN installed (although I am using an nVidia GPU).  I use Lasagne for LSTMs (using @craffel's fork of Lasagne) so, as far as I can tell, my code wouldn't use cuDNN even if it were installed.  Prior to commit b31511f01d32cc1bd93994fb3a2a4e93f52ac9f9 (specifically [this change](https://github.com/craffel/nntools/commit/b31511f01d32cc1bd93994fb3a2a4e93f52ac9f9#diff-185d143ba129327c0ce74dd75858134aR12)) I could `import lasagne` without requiring cuDNN.

(Great work with Lasagne though, it looks very promising!)

Just for completeness, here's the error I get:

```python
In [1]: import lasagne
Using gpu device 0: GeForce GTX 780 Ti
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-1-950fc8a00ce6> in <module>()
----> 1 import lasagne

/homes/dk3810/workspace/python/nntools/lasagne/__init__.py in <module>()
      5 import nonlinearities
      6 import init
----> 7 import layers
      8 import objectives
      9 import regularization

/homes/dk3810/workspace/python/nntools/lasagne/layers/__init__.py in <module>()
     11 from .corrmm import *
     12 from .recurrent import *
---> 13 from .dnn import *

/homes/dk3810/workspace/python/nntools/lasagne/layers/dnn.py in <module>()
     11 
     12 if not theano.config.device.startswith("gpu") or not dnn.dnn_available():
---> 13     raise ImportError("dnn not available")
     14 
     15 

ImportError: dnn not available
```

**Comment by benanne** [comment]
Are you absolutely sure you have the latest version of Lasagne? We just made it so that the submodules with external dependencies (cudnn, pylearn2) are no longer imported automatically when you `import lasagne`.

I did notice a while back that sometimes all submodules would get imported anyway, even if they are not explicitly imported in `__init__.py`. Maybe that's what's going on here? I have no explanation for this behavior though. If this is what's happening, we'll have to figure out how to reproduce it and hopefully fix it.
